addons/escoria-core/game/core-scripts/esc/esc_event_manager.gd

```diff
@@ -311,10 +311,13 @@ func queue_background_event(channel_name: String, event: ESCGrammarStmts.Event)
 #
 # #### Parameters
 # - exceptions: an optional list of events which should be left running or queued
-func interrupt(exceptions: PackedStringArray = []) -> void:
+# - stop_walking: boolean value (default true) determining whether the player 
+# (if any) has to be interrupted walking or not.
+func interrupt(exceptions: PackedStringArray = [], stop_walking = true) -> void:
 	if escoria.main.current_scene != null \
 			and escoria.main.current_scene.player != null \
-			and escoria.main.current_scene.player.is_moving():
+			and escoria.main.current_scene.player.is_moving() \
+			and stop_walking:
```

Comment:
This may seem a bit pedantic, but, what do you think about moving the `stop_walking` clause to the front of the predicate? That is, something like:

```
if stop_walking \
  and escoria.main.current_scene != null \
  and ...
```

...since it's a simple boolean and could short circuit the checks that involved more dereferencing.

What do you think?